torch/csrc/utils/schema_info.cpp

```diff
@@ -58,10 +58,43 @@ bool SchemaInfo::is_mutable(c10::string_view name) {
   return is_mutable(*index);
 }
 
+bool SchemaInfo::may_alias(
+    const c10::SchemaArgument& lhs,
+    const c10::SchemaArgument& rhs) {
+  bool basic_check = schema_.may_alias(lhs, rhs);
+  if (basic_check) {
+    return true;
+  }
+  if (!updated_) {
+    generateAliasMaps();
+  }
+  if (lhs.type == c10::SchemaArgType::input &&
+      rhs.type == c10::SchemaArgType::input) {
+    return input_alias_map_[lhs.index].count(rhs.index);
+  } else if (
+      lhs.type == c10::SchemaArgType::output &&
+      rhs.type == c10::SchemaArgType::output) {
+    for (size_t lhs_alias_input : output_alias_map_[lhs.index]) {
+      for (size_t rhs_alias_input : output_alias_map_[rhs.index]) {
+        if (lhs_alias_input == rhs_alias_input) {
+          return true;
+        }
+      }
```

Comment:
As these are maps, we should be able to just do `output_alias_map_[rhs.index].count(lhs_alias_input)` and eliminate the double for loop.